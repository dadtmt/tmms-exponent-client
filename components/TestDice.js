import React, { PropTypes } from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import RichTextDisplay from './RichTextDisplay'

const styleTestDice = (made, master) => {
  if (master) {
    return styles.master
  }
  if (made) {
    return styles.resolvedTestDice
  }

  return styles.TestDice
}

const TestDice = ({
  details,
  id,
  made,
  master,
  modifier,
  nbDices,
  nbSides,
  resolveTest,
  result,
  text
}) =>
<View style={styleTestDice(made, master)}>
  <TouchableOpacity onPress={() => { if(!master) { resolveTest({
    id
  })}}}>
    <RichTextDisplay rawContent={text} />
      <Text>Lancer les dés: {`${nbDices}D${nbSides}+${modifier}`}</Text>
      {
        (made || master) &&
          <View>
            <Text>Résultat: {result}</Text>
            <Text>Détails: {details}</Text>
          </View>
      }
  </TouchableOpacity>
</View>

TestDice.propTypes = {
  details: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  made: PropTypes.bool.isRequired,
  master: PropTypes.bool.isRequired,
  modifier: PropTypes.number.isRequired,
  nbDices: PropTypes.number.isRequired,
  nbSides: PropTypes.number.isRequired,
  resolveTest: PropTypes.func.isRequired,
  result: PropTypes.number.isRequired,
  text: PropTypes.object.isRequired
}

const styles = StyleSheet.create({
  TestDice: {
    backgroundColor: '#ccc',
    marginBottom: 10,
    padding: 5
  },
  master: {
    backgroundColor: '#f49242'
  },
  resolvedTestDice: {
    backgroundColor: '#0057e5',
    marginBottom: 10,
    padding: 5
  }
})


export default TestDice
