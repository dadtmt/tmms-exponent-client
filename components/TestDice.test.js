import React from 'react'
import renderer from 'react-test-renderer'

import TestDice from './TestDice'

it('renders without crashing', () => {
  const text = {
    blocks: [
      {
        data: {},
        depth: 0,
        entityRanges: [],
        inlineStyleRanges: [],
        key: '4bkmd',
        text: 'some rich text',
        type: 'unstyled'
      }
    ],
    entityMap: {}
  }
  const props = {
    details: 'details',
    id: 'SOME_ID',
    made: false,
    master: false,
    modifier: 0,
    nbDices: 1,
    nbSides: 6,
    resolveTest: jest.fn(),
    result: 6,
    text
  }
  expect(
    renderer.create(<TestDice {...props} />).toJSON()
  ).toMatchSnapshot()
})

it('renders made true', () => {
  const text = {
    blocks: [
      {
        data: {},
        depth: 0,
        entityRanges: [],
        inlineStyleRanges: [],
        key: '4bkmd',
        text: 'some rich text',
        type: 'unstyled'
      }
    ],
    entityMap: {}
  }
  const props = {
    details: 'details',
    id: 'SOME_ID',
    made: true,
    master: false,
    modifier: 0,
    nbDices: 1,
    nbSides: 6,
    resolveTest: jest.fn(),
    result: 6,
    text
  }
  expect(
    renderer.create(<TestDice {...props} />).toJSON()
  ).toMatchSnapshot()
})
