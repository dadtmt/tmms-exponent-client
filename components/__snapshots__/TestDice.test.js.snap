exports[`test renders made true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#0057e5",
      "marginBottom": 10,
      "padding": 5,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
          }
        }>
        some rich text
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Lancer les dés: 
      1D6+0
    </Text>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Résultat: 
        6
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Détails: 
        details
      </Text>
    </View>
  </View>
</View>
`;

exports[`test renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ccc",
      "marginBottom": 10,
      "padding": 5,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
          }
        }>
        some rich text
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Lancer les dés: 
      1D6+0
    </Text>
  </View>
</View>
`;
